yarn-project/archiver/src/archiver/kv_archiver_store/log_store.ts

```diff
@@ -105,6 +142,21 @@ export class LogStore {
     }
   }
 
+  getLogsByTags(tags: Fr[]): Promise<EncryptedL2NoteLog[][]> {
+    return this.db.transaction(() => {
+      return tags.map(tag => {
+        const logHashes = Array.from(this.#noteEncryptedLogHashesByTag.getValues(tag.toString()));
+        return (
+          logHashes
+            .map(hash => this.#noteEncryptedLogsByHash.get(hash))
+            // filter out undefined values, since we should never store the hashes of non-existing logs (the addLogs transaction ensures this)
+            .filter(noteLogBuffer => noteLogBuffer)
```

Comment:
Should we maybe then remove the filter and have `fromBuffer` below fail if an undefined value makes its way there?

Reply:
I did it like this so we could recover from db corruption instead of having some cursed logs that always fail. I can change it, but we would then have that risk.